invenio_app_ils/ui/src/invenio_app_ils/common/api/eitems/eitem.js

```diff
@@ -0,0 +1,49 @@
+import { http } from '../base';
+import { serializer } from './serializer';
+import { prepareSumQuery } from '../utils';
+
+const eitemURL = '/eitems/';
+
+const get = async eitemPid => {
+  const response = await http.get(`${eitemURL}${eitemPid}`);
+  response.data = serializer.fromJSON(response.data);
+  return response;
+};
+
+const list = async query => {
+  const response = await http.get(`${eitemURL}?q=${query}`);
+  response.data.total = response.data.hits.total;
+  response.data.hits = response.data.hits.hits.map(hit =>
+    serializer.fromJSON(hit)
+  );
+  return response;
+};
+
+class QueryBuilder {
```

Comment:
Ok, I see now the purpose of the aforementioned tests, would it be possible to extract the query Builder from `api/item.js` as stand alone functionality, decorated with relevant props from the other apis, and then use of Query Builder for building queries, instead of defining a specific one for each model?

Reply:
This could be a possible refactor yes :)